lib/open-search-client.ts

```diff
@@ -0,0 +1,378 @@
+/* eslint-disable camelcase */
+/* eslint-disable no-underscore-dangle */
+import got4aws from 'got4aws';
+import { get } from 'lodash';
+import { percolateResponseSchema } from './schemas';
+
+import {
+  CloseScrollResponse,
+  DocumentItem,
+  DocumentItemTerm,
+  PutDocumentItemResponse,
+} from './open-search-types';
+
+/**
+ * @param prefixUrl
+ * @returns
+ */
+const gotEs = (prefixUrl: string) => got4aws().extend({
+  prefixUrl,
+  responseType: 'json',
+  resolveBodyOnly: true,
+});
+
+export interface OpenScrollOptions {
+  includes?: string[]
+  scrollTimeout?: number,
+  size?: number
+}
+
+/**
+ * Queries OpenSearch for all items and returns them while also
+ * opening a scroll.
+ * See https://opensearch.org/docs/latest/opensearch/rest-api/scroll/
+ * for more details.
+ *
+ * @param prefixUrl - Open Search endpoint.
+ * @param index - Name of the index to query.
+ * @param options - Options to the request.
+ * @param {string[]} [options.includes=['*']] -
+ *
+ * @returns Open Search query results including a
+ * scroll ID.
+ */
+export const openScroll = async (
+  prefixUrl: string,
+  index: string,
+  options: OpenScrollOptions = {}
+): Promise<unknown> => {
+  const {
+    includes = ['*'],
+    scrollTimeout = 60,
+    size = 500,
+  } = options;
+
+  // eslint-disable-next-line no-return-await
+  return await gotEs(prefixUrl).post(
+    `${index}/_search`,
+    {
+      json: {
+        scroll: `${scrollTimeout}m`,
+        _source: {
+          includes,
+        },
+        size,
+      },
+    }
+  );
+};
+
+/**
+ * Returns search results for an open Open Search scroll.
+ * See https://opensearch.org/docs/latest/opensearch/rest-api/scroll/
+ * for more details.
+ *
+ * @param prefixUrl - Open Search endpoint.
+ * @param scrollId - Scroll ID of the open scroll.
+ *
+ * @returns Open Search query results including scroll ID. The scroll ID
+ * in this response should always be used in future next scroll requests.
+ */
+export const nextScroll = async (
+  prefixUrl: string,
+  scrollId: string,
+  scrollTimeout = 60
+): Promise<unknown> => gotEs(prefixUrl).post(
+  '_search/scroll',
+  {
+    json: {
+      scroll: `${scrollTimeout}m`,
+      scroll_id: scrollId,
+    },
+  }
+);
+
+/**
+ * Closes an open Open Search scroll query.
+ * See https://opensearch.org/docs/latest/opensearch/rest-api/scroll/
+ * for more details.
+ *
+ * @param prefixUrl - Open Search endpoint.
+ * @param scrollId - Scroll ID of the open scroll.
+ *
+ * @returns Open Search close scroll response.
+ */
+export const closeScroll = (
+  prefixUrl: string,
+  scrollId: string
+): Promise<CloseScrollResponse> => got4aws().delete(
```

Comment:
Since this has a type if I use the gotES I get a TypeScript error that I think is due to it not seeing that we're resolving the JSON body.